packages/stores/src/queries/pools/cfmm-to-cl-link.ts

```diff
@@ -0,0 +1,63 @@
+import { KVStore } from "@keplr-wallet/common";
+import {
+  ChainGetter,
+  ObservableChainQuery,
+  ObservableChainQueryMap,
+} from "@keplr-wallet/stores";
+import { computed, makeObservable } from "mobx";
+
+export class ObservableQueryCfmmToConcentratedLiquidityPoolLink extends ObservableChainQuery<{
+  concentrated_pool_id: string;
+}> {
+  constructor(
+    kvStore: KVStore,
+    chainId: string,
+    chainGetter: ChainGetter,
+    cfmmPoolId: string
+  ) {
+    super(
+      kvStore,
+      chainId,
+      chainGetter,
+      `/osmosis/gamm/v1beta1/concentrated_pool_id_link_from_cfmm/${cfmmPoolId}`
+    );
+
+    makeObservable(this);
+  }
+
+  /** Returns the ID of the linked concentrated liquidity pool if it exists, `false` if there's no link, or `undefined` if the request is in flight. */
+  @computed
+  get concentratedLiquidityPoolId() {
+    // link doesn't exist
+    if (this.response?.status === 400) {
```

Comment:
is 400 the only possible error code we want to handle?

Reply:
That's the specific expected code that the endpoint will return if there's confirmed to be no ID for the given CFMM pool ID. Other errors are unexpected.